projects/js-packages/charts/src/components/pie-semi-circle-chart/test/pie-semi-circle-chart.test.tsx

```diff
@@ -71,11 +71,10 @@ describe( 'PieSemiCircleChart', () => {
 			await user.hover( firstSegment );
 		} );
 
-		// Check for tooltip content with flexible text matching
-		const tooltipText = screen.getByText( content => {
-			return content.includes( 'MacOS' ) || content.includes( '30K' );
-		} );
-		expect( tooltipText ).toBeInTheDocument();
+		// Check for tooltip by looking for the specific tooltip role
+		const tooltip = screen.getByRole( 'tooltip' );
+		expect( tooltip ).toHaveTextContent( 'MacOS' );
+		expect( tooltip ).toHaveTextContent( '30K' );
 	} );
```

Comment:
I updated tooltip tests to use `getByRole('tooltip')` instead of text-based queries to avoid conflict with legend text when legends are enabled